core/src/test/java/org/apache/iceberg/TestIncrementalDataTableScan.java

```diff
@@ -78,8 +81,28 @@ public void testAppends() {
     add(table.newAppend(), files("C"));
     add(table.newAppend(), files("D"));
     add(table.newAppend(), files("E")); // 5
+
+    class MyListener implements Listener<IncrementalScanEvent> {
+
+      IncrementalScanEvent lastEvent = null;
+
+      public void notify(IncrementalScanEvent event) {
+        this.lastEvent = event;
+      }
+
+      public IncrementalScanEvent event() {
+        return lastEvent;
+      }
+    }
+
+    MyListener listener1 = new MyListener();
+    Listeners.register(listener1, IncrementalScanEvent.class);
     filesMatch(Lists.newArrayList("B", "C", "D", "E"), appendsBetweenScan(1, 5));
+    Assert.assertTrue(listener1.event().fromSnapshotId() == 1);
```

Comment:
I'm in favor of check all fields as it's the only test verifying IncrementalScanEvent, but others may have different thought on this.

Reply:
Thanks @HeartSaVioR!  I have updated the test to fetch and check all the fields of this event.